src/Simplify.elm

```diff
@@ -6370,6 +6383,89 @@ indexAccessChecks collection checkInfo n =
                 Nothing
 
 
+setChecks : CollectionProperties (FromListProperties (IndexableProperties {})) -> CheckInfo -> Maybe (Error {})
+setChecks collection checkInfo =
+    firstThatConstructsJust
+        [ \() ->
+            Maybe.andThen
+                (\emptiableArg -> callOnEmptyReturnsEmptyCheck emptiableArg collection checkInfo)
+                (thirdArg checkInfo)
+        , \() ->
+            case Evaluate.getInt checkInfo checkInfo.firstArg of
+                Just n ->
+                    case secondArg checkInfo of
+                        Just replacementArg ->
+                            if n < 0 then
+                                Just
+                                    (alwaysReturnsLastArgError
+                                        (qualifiedToString checkInfo.fn ++ " with negative index")
+                                        { lastArg = thirdArg checkInfo
+                                        , lastArgRepresents = collection.represents
+                                        }
+                                        checkInfo
+                                    )
+
+                            else
+                                setOnKnownElementChecks collection checkInfo n (Node.range replacementArg)
+
+                        Nothing ->
+                            Just
+                                (Rule.errorWithFix
+                                    { message = qualifiedToString checkInfo.fn ++ " with negative index will always return the same given " ++ collection.represents
+                                    , details =
+                                        [ "You can replace this call by \\_ -> identity." ]
+                                    }
+                                    checkInfo.fnRange
+                                    [ Fix.replaceRangeBy checkInfo.parentRange "\\_ -> identity" ]
+                                )
+
+                Nothing ->
+                    Nothing
+        ]
+        ()
+
+
+setOnKnownElementChecks :
+    CollectionProperties (FromListProperties (IndexableProperties {}))
+    -> CheckInfo
+    -> Int
+    -> Range
+    -> Maybe (Error {})
+setOnKnownElementChecks collection checkInfo n replacementArgRange =
+    case thirdArg checkInfo of
+        Just collectionArg ->
+            case collection.literalElements checkInfo.lookupTable collectionArg of
+                Just literalElements ->
+                    case List.drop n literalElements |> List.head of
+                        Just element ->
+                            Just
+                                (Rule.errorWithFix
+                                    { message = "The element returned by " ++ qualifiedToString checkInfo.fn ++ " is known"
```

Comment:
This message seems to be copied from get but it is both incorrect and very undescriptive I'd say.